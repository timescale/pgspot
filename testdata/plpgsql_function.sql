
-- unsafe call inside function
CREATE FUNCTION plpgsqlfunc3() RETURNS TEXT LANGUAGE PLPGSQL AS $$ BEGIN SELECT unsafe_call3(); END; $$;
CREATE FUNCTION plpgsqlfunc4() RETURNS TEXT LANGUAGE PLPGSQL AS $$ BEGIN PERFORM unsafe_call4(); END; $$;
CREATE FUNCTION plpgsqlfunc5() RETURNS TEXT LANGUAGE PLPGSQL AS $$ BEGIN RETURN unsafe_call5(); END; $$;

-- safe call inside function
CREATE FUNCTION plpgsqlfunc7() RETURNS TEXT LANGUAGE PLPGSQL AS $$ BEGIN SELECT safe_call7(); END; $$ SET search_path TO 'pg_catalog';
CREATE FUNCTION plpgsqlfunc8() RETURNS TEXT LANGUAGE PLPGSQL AS $$ BEGIN PERFORM safe_call8(); END; $$ SET search_path TO 'pg_catalog';

-- unsafe call inside function, since search_path for body is not safe
SET search_path TO pg_catalog;
CREATE FUNCTION plpgsqlfunc13() RETURNS TEXT LANGUAGE PLPGSQL AS $$ BEGIN SELECT unsafe_call13(); END; $$;

CREATE FUNCTION plpgsqlfunc16() RETURNS TEXT LANGUAGE PLPGSQL AS $$
BEGIN
  ASSERT true;
  SELECT unsafe_call17();
  SET search_path TO pg_catalog;
  SELECT safe_call19();
  RESET search_path;
  SELECT unsafe_call21();
  SELECT foo.safe_call22();
END;
$$;
