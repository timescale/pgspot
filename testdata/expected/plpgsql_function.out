PS005: Function without explicit search_path: plpgsqlfunc3() at line 2
PS016: Unqualified function call: unsafe_call3 at line 2
PS005: Function without explicit search_path: plpgsqlfunc4() at line 4
PS016: Unqualified function call: unsafe_call4 at line 4
PS005: Function without explicit search_path: plpgsqlfunc5() at line 5
PS016: Unqualified function call: unsafe_call5 at line 5
PS005: Function without explicit search_path: plpgsqlfunc13() at line 13
PS016: Unqualified function call: unsafe_call13 at line 13
PS005: Function without explicit search_path: plpgsqlfunc16() at line 14
PS016: Unqualified function call: unsafe_call17 at line 14
PS016: Unqualified function call: unsafe_call21 at line 14

 Errors: 0 Warnings: 11 Unknown: 0 

